flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/RowDataElementConverter.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.flink.connector.dynamodb.table;
+
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.connector.dynamodb.sink.DynamoDbWriteRequest;
+import org.apache.flink.connector.dynamodb.sink.DynamoDbWriteRequestType;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.DataType;
+
+/** Implementation of an {@link ElementConverter} specific for DynamoDb sink. */
+public class RowDataElementConverter implements ElementConverter<RowData, DynamoDbWriteRequest> {
+
+    private final DataType physicalDataType;
+    private final RowDataToAttributeValueConverter rowDataToAttributeValueConverter;
+
+    public RowDataElementConverter(DataType physicalDataType) {
+        this.physicalDataType = physicalDataType;
+        this.rowDataToAttributeValueConverter =
+                new RowDataToAttributeValueConverter(physicalDataType);
+    }
+
+    @Override
+    public DynamoDbWriteRequest apply(RowData element, SinkWriter.Context context) {
+        switch (element.getRowKind()) {
+            case INSERT:
+            case UPDATE_AFTER:
+                return getPutRequest(element);
+            case UPDATE_BEFORE:
+            case DELETE:
+                return getDeleteRequest(element);
+            default:
+                throw new TableException("Unsupported message kind: " + element.getRowKind());
+        }
+    }
+
+    private DynamoDbWriteRequest getPutRequest(RowData row) {
+        return DynamoDbWriteRequest.builder()
+                .setType(DynamoDbWriteRequestType.PUT)
+                .setItem(rowDataToAttributeValueConverter.convertRowData(row))
+                .build();
+    }
+
+    private DynamoDbWriteRequest getDeleteRequest(RowData row) {
+        return DynamoDbWriteRequest.builder()
+                .setType(DynamoDbWriteRequestType.DELETE)
+                .setItem(rowDataToAttributeValueConverter.convertRowData(row))
+                .build();
+    }
```

Comment:
nit: I like this, however you could create the builder in the previous method

```
@Override
    public DynamoDbWriteRequest apply(RowData element, SinkWriter.Context context) {
      var builder = DynamoDbWriteRequest.builder()
                .setItem(rowDataToAttributeValueConverter.convertRowData(row));
    
        switch (element.getRowKind()) {
            case INSERT:
            case UPDATE_AFTER:
                builder.setType(DynamoDbWriteRequestType.PUT);
            case UPDATE_BEFORE:
            case DELETE:
                builder.setType(DynamoDbWriteRequestType.DELETE);
            default:
                throw new TableException("Unsupported message kind: " + element.getRowKind());
        }

      return builder.build();
    }
```

Reply:
Ok, done!